### ISSUE TITLE
New operator is not wholly supported

### ISSUE THREAD

**Issue by rossm6** [6utt3rfly/jse-eval#70]
I've noticed something like this doesn't work.

```
const { evaluate, parse, registerPlugin }= require('jse-eval');
const newPlugin = require('@jsep-plugin/new');
registerPlugin(newPlugin);
const p = parse("new Date(new Date().setDate(new Date().getDate()-5))");
evaluate(p, { Date }); // gives error: Error: 'setDate' is not a function
```

**Comment by 6utt3rfly** [comment]
Thanks for reporting @rossm6 ! It looks like an incorrect AST is being generated by JSEP (and I see the [related issue](https://github.com/EricSmekens/jsep/issues/243) you submitted there).

**Comment by 6utt3rfly** [comment]
closed in [jsep-plugin/new v1.0.3](https://github.com/EricSmekens/jsep/issues/243#issuecomment-1421336025)
